### PR TITLE
[SMALLFIX] POM cleanup

### PR DESCRIPTION
**Pull request by apc999**
removing unused dependencies and making dependencies explicit.

### CODE REVIEW

File: underfs/hdfs/pom.xml [hunk 1]
```diff
@@ -69,7 +69,7 @@
         <dependency>
           <groupId>org.apache.hadoop</groupId>
           <artifactId>hadoop-core</artifactId>
-          <version>1.0.4</version>
+          <version>${hadoop.version}</version>
```
Comment (calvinjia) [inline underfs/hdfs/pom.xml]: Could this be causing the test issues?

File: core/server/common/pom.xml [hunk 2]
```diff
@@ -33,20 +33,49 @@
...
     <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-databind</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
```
Comment (jsimsa) [inline core/server/common/pom.xml]: nit: sort lexicographically (group, artifact)

Comment (jsimsa) [inline core/server/common/pom.xml]: so
1. jackson-core
2. jackson-databind
3. guava

File: core/server/common/pom.xml [hunk 3]
```diff
@@ -33,20 +33,49 @@
...
+      <artifactId>javax.servlet-api</artifactId>
+    </dependency>
+    <dependency>
       <groupId>javax.ws.rs</groupId>
       <artifactId>javax.ws.rs-api</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.eclipse.jetty</groupId>
-      <artifactId>jetty-annotations</artifactId>
+      <groupId>org.apache.thrift</groupId>
```
Comment (jsimsa) [inline core/server/common/pom.xml]: fix order

File: core/server/master/pom.xml [hunk 4]
```diff
@@ -33,26 +33,63 @@
...
     <dependency>
-      <groupId>com.fasterxml.jackson.core</groupId>
-      <artifactId>jackson-databind</artifactId>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
     </dependency>
     <dependency>
-      <groupId>com.google.code.findbugs</groupId>
-      <artifactId>annotations</artifactId>
+      <groupId>commons-lang</groupId>
```
Comment (jsimsa) [inline core/server/master/pom.xml]: `commons` should not be between two `com` ... I think it should go before

File: core/server/proxy/pom.xml [hunk 5]
```diff
@@ -33,14 +33,55 @@
...
+    </dependency>
+    <dependency>
       <groupId>org.glassfish.jersey.containers</groupId>
       <artifactId>jersey-container-servlet-core</artifactId>
     </dependency>
     <dependency>
+      <groupId>org.glassfish.jersey.core</groupId>
+      <artifactId>jersey-server</artifactId>
+    </dependency>
+    <!-- Needed at runtime -->
```
Comment (jsimsa) [inline core/server/proxy/pom.xml]: "Used by reflection" for consistency

File: minicluster/pom.xml [hunk 6]
```diff
@@ -34,14 +34,17 @@
...
+    <dependency>
       <groupId>org.apache.curator</groupId>
       <artifactId>curator-test</artifactId>
       <scope>compile</scope>
     </dependency>
     <dependency>
-      <groupId>org.powermock</groupId>
-      <artifactId>powermock-reflect</artifactId>
-      <scope>compile</scope>
+      <groupId>org.apache.commons</groupId>
```
Comment (jsimsa) [inline minicluster/pom.xml]: fix order

File: pom.xml [hunk 7]
```diff
@@ -699,10 +732,10 @@
   <!-- Dependencies to add to all projects. -->
   <dependencies>
     <!-- Dependencies in the compile scope. -->
-    <!-- Thread-safety annotations. -->
+    <!-- FindBugs amd concurrency annotations. -->
```
Comment (jsimsa) [inline pom.xml]: s/amd/and

File: underfs/hdfs/pom.xml (hunk 1 above)
Comment (apc999) [inline underfs/hdfs/pom.xml]: I think it is different. https://github.com/Alluxio/alluxio/pull/5408/files aims to fix the test issue

File: core/server/common/pom.xml (hunk 2 above)
Comment (apc999) [inline core/server/common/pom.xml]: done

File: core/server/common/pom.xml (hunk 3 above)
Comment (apc999) [inline core/server/common/pom.xml]: done

File: core/server/master/pom.xml (hunk 4 above)
Comment (apc999) [inline core/server/master/pom.xml]: done

File: minicluster/pom.xml (hunk 6 above)
Comment (apc999) [inline minicluster/pom.xml]: done

File: pom.xml (hunk 7 above)
Comment (apc999) [inline pom.xml]: done

**Comment by apc999** [comment]
result of `mvn dependency:analyze`. A few exceptions:

- We globally depend on `com.google.code.findbugs:annotations:jar` which is a superset of `com.google.code.findbugs:jsr305:jar` for findbugs and concurrency annotations. However, the global dependency (from parent pom) has lower priority than defined external dependencies. So sometimes, from `mvn dependency:analyze` it shows `com.google.code.findbugs:jsr305:jar` is used but undeclared. 

- For `core-client-hdfs`, we depend on `hadoop-client` due to compatibility across versions.

```
[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-core-protobuf ---
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.code.findbugs:annotations:jar:3.0.1:provided
[WARNING]    org.slf4j:slf4j-api:jar:1.7.2:compile
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    junit:junit:jar:4.12:test
[WARNING]    org.mockito:mockito-all:jar:1.10.8:test
[WARNING]    org.powermock:powermock-api-mockito:jar:1.6.1:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-module-junit4:jar:1.6.1:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-core-common ---
[WARNING] Used undeclared dependencies found:
[WARNING]    com.google.code.findbugs:jsr305:jar:1.3.9:provided
[WARNING] Unused declared dependencies found:
[WARNING]    io.dropwizard.metrics:metrics-json:jar:3.1.2:compile
[WARNING]    com.google.code.findbugs:annotations:jar:3.0.1:provided

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-underfs-gcs ---
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    org.powermock:powermock-api-mockito:jar:1.6.1:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-module-junit4:jar:1.6.1:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-underfs-glusterfs ---
[WARNING] Used undeclared dependencies found:
[WARNING]    com.google.code.findbugs:jsr305:jar:1.3.9:compile
[WARNING] Unused declared dependencies found:
[WARNING]    org.gluster:glusterfs-hadoop:jar:2.3.13:compile
[WARNING]    com.google.code.findbugs:annotations:jar:3.0.1:provided
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    org.mockito:mockito-all:jar:1.10.8:test
[WARNING]    org.powermock:powermock-api-mockito:jar:1.6.1:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-module-junit4:jar:1.6.1:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-underfs-hdfs ---
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    org.mockito:mockito-all:jar:1.10.8:test
[WARNING]    org.powermock:powermock-api-mockito:jar:1.6.1:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-module-junit4:jar:1.6.1:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-underfs-local ---
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    org.mockito:mockito-all:jar:1.10.8:test
[WARNING]    org.powermock:powermock-api-mockito:jar:1.6.1:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-module-junit4:jar:1.6.1:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-underfs-oss ---
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-underfs-s3 ---
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    org.powermock:powermock-api-mockito:jar:1.6.1:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-module-junit4:jar:1.6.1:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-underfs-s3a ---
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-underfs-swift ---
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-core-client-fs ---
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-core-client-hdfs ---
[WARNING] Used undeclared dependencies found:
[WARNING]    org.apache.hadoop:hadoop-mapreduce-client-core:jar:2.2.0:compile
[WARNING]    org.apache.hadoop:hadoop-common:jar:2.2.0:compile
[WARNING] Unused declared dependencies found:
[WARNING]    org.apache.hadoop:hadoop-client:jar:2.2.0:compile
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-core-client-runtime ---
[WARNING] Unused declared dependencies found:
[WARNING]    org.alluxio:alluxio-core-client-hdfs:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-core-client-fs:jar:1.5.0-SNAPSHOT:compile
[WARNING]    com.google.code.findbugs:annotations:jar:3.0.1:provided
[WARNING]    org.slf4j:slf4j-api:jar:1.7.2:compile
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    junit:junit:jar:4.12:test
[WARNING]    org.mockito:mockito-all:jar:1.10.8:test
[WARNING]    org.powermock:powermock-api-mockito:jar:1.6.1:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-module-junit4:jar:1.6.1:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-core-server-common ---
[WARNING] Unused declared dependencies found:
[WARNING]    org.eclipse.jetty:jetty-jsp:jar:9.2.16.v20160414:compile
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-module-junit4:jar:1.6.1:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-core-server-master ---
[WARNING] Used undeclared dependencies found:
[WARNING]    com.google.code.findbugs:jsr305:jar:1.3.9:provided
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    org.powermock:powermock-api-mockito:jar:1.6.1:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-module-junit4:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-core-server-proxy ---
[WARNING] Unused declared dependencies found:
[WARNING]    org.glassfish.jersey.media:jersey-media-json-jackson:jar:2.22:compile
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    org.powermock:powermock-api-mockito:jar:1.6.1:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-module-junit4:jar:1.6.1:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-core-server-worker ---
[WARNING] Unused declared dependencies found:
[WARNING]    org.alluxio:alluxio-underfs-local:jar:1.5.0-SNAPSHOT:test
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-shell ---
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    junit:junit:jar:4.12:test
[WARNING]    org.mockito:mockito-all:jar:1.10.8:test
[WARNING]    org.powermock:powermock-api-mockito:jar:1.6.1:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-module-junit4:jar:1.6.1:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-minicluster ---
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    junit:junit:jar:4.12:test
[WARNING]    org.mockito:mockito-all:jar:1.10.8:test
[WARNING]    org.powermock:powermock-api-mockito:jar:1.6.1:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-module-junit4:jar:1.6.1:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test

[INFO] --- maven-dependency-plugin:2.8:analyze (default-cli) @ alluxio-assemblies ---
[WARNING] Unused declared dependencies found:
[WARNING]    org.alluxio:alluxio-core-server-master:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-core-server-proxy:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-core-server-worker:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-examples:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-keyvalue-server:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-shell:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-underfs-gcs:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-underfs-glusterfs:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-underfs-hdfs:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-underfs-local:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-underfs-oss:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-underfs-s3:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-underfs-s3a:jar:1.5.0-SNAPSHOT:compile
[WARNING]    org.alluxio:alluxio-underfs-swift:jar:1.5.0-SNAPSHOT:compile
[WARNING]    com.google.code.findbugs:annotations:jar:3.0.1:provided
[WARNING]    org.slf4j:slf4j-api:jar:1.7.2:compile
[WARNING]    com.google.guava:guava-testlib:jar:19.0:test
[WARNING]    junit:junit:jar:4.12:test
[WARNING]    org.mockito:mockito-all:jar:1.10.8:test
[WARNING]    org.powermock:powermock-api-mockito:jar:1.6.1:test
[WARNING]    org.powermock:powermock-core:jar:1.6.1:test
[WARNING]    org.powermock:powermock-module-junit4:jar:1.6.1:test
[WARNING]    org.powermock:powermock-reflect:jar:1.6.1:test
```

**Comment by apc999** [comment]
@calvinjia @jsimsa  comments addressed. PTAL

**Pull request merged**